File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples/VB/WPFListBox-VB/WPFListBox/Components/AppConfig.vb
Imports System.Configuration

Public Class AppConfig
  Protected Shared Function GetConnectString(ByVal key As String) As String
    Return ConfigurationManager.ConnectionStrings(key).ConnectionString
  End Function

  Public Shared ReadOnly Property ConnectString() As String
    Get
      Return GetConnectString("WPFComponents.Properties.Settings.AdventureWorksLTConnectionString")
    End Get
  End Property
End Class

File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples-2012/CSharp/WPFListBox/WPFListBox/Components/AppConfig.cs
using System;
using System.Configuration;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace WPFListBox
{
  public class AppConfig
  {
    protected static string GetConnectString(string key)
    {
      return ConfigurationManager.ConnectionStrings[key].ConnectionString;
    }

    public static string ConnectString
    {
      get { return GetConnectString("WPFComponents.Properties.Settings.AdventureWorksLTConnectionString"); }
    }
  }
}
